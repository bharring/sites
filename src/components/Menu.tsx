import { IonContent, IonIcon, IonItem, IonLabel, IonList, IonListHeader, IonMenu, IonMenuToggle } from '@ionic/react';
import { logOutOutline } from 'ionicons/icons';
import React from 'react';

import { UserContext } from '../state/userContext';

const Menu: React.FC = () => {
  const { signOut } = React.useContext(UserContext);

  return (
    <IonMenu contentId="main-content">
      <IonContent>
        <IonList lines="inset">
          <IonListHeader lines="inset">
            <IonLabel>
              <h1>Header</h1>
            </IonLabel>
          </IonListHeader>

          <IonMenuToggle auto-hide="false">
            <IonItem data-testid="button" onClick={signOut} button>
              <IonIcon slot="start" icon={logOutOutline}></IonIcon>
              <IonLabel>Log Out</IonLabel>
            </IonItem>
          </IonMenuToggle>
        </IonList>
      </IonContent>
    </IonMenu>
  );
};
export default Menu;
