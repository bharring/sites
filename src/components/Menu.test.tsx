import '@testing-library/jest-dom';

import { cleanup, fireEvent, render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

import Menu from './Menu';

describe('<Menu />', () => {
  afterEach(cleanup);
  test('renders Menu snapshot', () => {
    const tree = renderer.create(<Menu />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('clicking Logout item calls logout()', () => {
    const signOutMock = jest.fn();
    jest.spyOn(React, 'useContext').mockImplementation(() => ({ signOut: signOutMock }));

    const { getByTestId } = render(<Menu />);

    expect(signOutMock).not.toHaveBeenCalled();
    fireEvent.click(getByTestId('button'));
    expect(signOutMock).toHaveBeenCalled();
  });
});
