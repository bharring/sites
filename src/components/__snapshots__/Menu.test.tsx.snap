// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> renders Menu snapshot 1`] = `
<ion-menu
  content-id="main-content"
>
  <ion-content>
    <ion-list
      lines="inset"
    >
      <ion-list-header
        lines="inset"
      >
        <ion-label>
          <h1>
            Header
          </h1>
        </ion-label>
      </ion-list-header>
      <ion-menu-toggle
        auto-hide="false"
      >
        <ion-item
          button={true}
          data-testid="button"
          onClick={[Function]}
        >
          <ion-icon
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Log Out</title><path d='M304 336v40a40 40 0 01-40 40H104a40 40 0 01-40-40V136a40 40 0 0140-40h152c22.09 0 48 17.91 48 40v40M368 336l80-80-80-80M176 256h256' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
            slot="start"
          />
          <ion-label>
            Log Out
          </ion-label>
        </ion-item>
      </ion-menu-toggle>
    </ion-list>
  </ion-content>
</ion-menu>
`;
